### PR TITLE
stargate: split types/

### PR DESCRIPTION
**Pull request by fedekunze**
- stargate: split proto and base files
- stargate: split app/ + codec/
- stargate: split types/

### CODE REVIEW

File: types/account.go [hunk 1]
```diff
@@ -120,14 +79,13 @@ func (acc EthAccount) MarshalYAML() (interface{}, error) {
 func (acc EthAccount) MarshalJSON() ([]byte, error) {
 	var ethAddress = ""
 
-	if acc.BaseAccount != nil && acc.Address != nil {
+	if acc.BaseAccount != nil && acc.Address == "" {
```
Comment (freddyli7) [inline types/account.go]: not sure if this `==` is a typo, but seems that condition here has been changed to the opposit way.

**Comment by freddyli7** [comment]
added a single comment

File: types/account.go [hunk 2]
```diff
@@ -3,97 +3,56 @@ package types
...
 	ethcmn "github.com/ethereum/go-ethereum/common"
 	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 )
 
-var _ exported.Account = (*EthAccount)(nil)
-var _ exported.GenesisAccount = (*EthAccount)(nil)
-
-func init() {
-	authtypes.RegisterAccountTypeCodec(&EthAccount{}, EthAccountName)
-}
```
Comment (araskachoi) [inline types/account.go:25]: do we not need to init and run `RegisterAccounTypeCodec()` anymore?

File: types/account.go [hunk 3]
```diff
@@ -3,97 +3,56 @@ package types
...
-	if err := acc.SetCoins(coins); err != nil {
-		panic(fmt.Errorf("could not set %s coins for address %s: %w", denom, acc.EthAddress().String(), err))
-	}
+	return ethcmn.BytesToAddress(acc.GetAddress().Bytes())
 }
 
 type ethermintAccountPretty struct {
-	Address       sdk.AccAddress `json:"address" yaml:"address"`
-	EthAddress    string         `json:"eth_address" yaml:"eth_address"`
-	Coins         sdk.Coins      `json:"coins" yaml:"coins"`
```
Comment (araskachoi) [inline types/account.go:84]: just confirming, the coin is now stored in the bank module and not on the account,?

File: types/account.go (hunk 2 above)
Comment (fedekunze) [reply types/account.go:25]: no, we don't use amino

File: types/account.go (hunk 3 above)
Comment (fedekunze) [reply types/account.go:84]: yup

File: types/errors.go [hunk 4]
```diff
@@ -20,4 +20,10 @@ var (
 
 	// ErrVMExecution returns an error resulting from an error in EVM execution.
 	ErrVMExecution = sdkerrors.Register(RootCodespace, 4, "error while executing evm transaction")
+
+	// ErrMarshalBigInt returns an error resulting from marshaling a big.Int from a string.
+	ErrMarshalBigInt = sdkerrors.Register(RootCodespace, 5, "cannot unmarshal big.Int from string")
```
Comment (araskachoi) [inline types/errors.go:25]: edit the error message to `"cannot marshal big.Int from string"`

**Review by araskachoi — APPROVED** [review]

**Comment by araskachoi** [comment]
looks good, just some minor change requests and questions!

File: types/codec.go [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
-	EthAccountName = "ethermint/EthAccount"
-)
-
-// RegisterCodec registers the account interfaces and concrete types on the
-// provided Amino codec.
-func RegisterCodec(cdc *codec.Codec) {
-	cdc.RegisterConcrete(&EthAccount{}, EthAccountName, nil)
+// RegisterInterfaces registers the tendermint concrete client-related
+// implementations and interfaces.
+func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
```
[the remark this answers is not in the document]
Comment (fedekunze) [reply types/codec.go:10]: Highly recommend reading this: https://docs.cosmos.network/master/core/encoding.html#gogoproto

**Review by noot — APPROVED** [review]

**Pull request closed without merging**